code/0010-define-calculations-of-existential-deposit.md

```diff
@@ -0,0 +1,149 @@
+extensibility in the future.
+
+## Requirements
+
+* ED values for all tokens supported by the chain are available AND consistent
+  across our runtime, node, and front-end applications.
+
+* When the ED of token A is swapped into token B, the new balance of token B is
+  substantial for ED. In other words, a user cannot "delete" their balance when 
+  they convert it into another token.
+
+* The implementation of our ED value retrieval should allow for the overwriting
+  of the standard ED value defined in this document.
+
+## Method
+
+### Initial Standard ED Value
+
+With the following definitions: 
+
+* `local_ed` - The functional ED of a asset on our chain.
+
+* `native_token_ed` - The ED of our chains native asset.
+
+* `foreign_asset_ratio` - The ratio of 'one in the foreign asset' over 'one in 
+our native asset'
+
+The ED of any asset should be defined as follows:
+```
+local_ed = native_token_ed * foreign_asset_ratio
+```
+
+This method of setting ED ensures requirement 2.
+
+Example
+```
+token_a_local_ed = 100_000_000
+token_a_asset_ratio = Ratio(1_000_000_000, 1_000_000_000_000)
+token_b_local_ed = 100_000
+token_a_asset_ratio = Ratio(1_000_000, 1_000_000_000_000)
+native_token_ed = 100_000_000_000
+
+fn native_to_token(native_balance, token_asset_ratio) {
+  native_balance * token_asset_ratio
+}
+
+fn token_to_native(token_balance, token_asset_ratio) {
+  token_balance / token_asset_ratio
+}
+
+token_a_ed_as_native = token_to_native(token_a_local_ed, token_a_asset_ratio)
+token_b_ed_as_native = token_to_native(token_b_local_ed, token_b_asset_ratio)
+native_as_token_b = native_to_token(native_token_ed, token_b_asset_ratio)
+token_a_as_token_b = native_to_token(token_a_ed_as_native, token_b_asset_ratio)
+
+assert(token_a_ed_as_native == native_token_ed)
+assert(token_b_ed_as_native == native_token_ed)
+assert(token_a_ed_as_native == token_b_ed_as_native)
+assert(native_as_token_b == token_a_as_token_b)
+```
+
+### Overwriting & Asset Registry Storage
+
+For a token to have its ED be correctly calculated, and therefore to exists on 
+chain, it should have its `foreign_asset_ratio` defined and stored.
+
+If the need arises, an optional ED value can be provided to an asset within
+Asset Registry. This will have priority over the default ED calculation and will
+be evaluated as is.
+
+## Implementation
+
+### Updates to the Functionality of `multi_existential_deposits`
+
+The current implementation of `multi_existential_deposits` does the following:
+  
+  1. If an ED is defined for an asset in Asset Registry, return the ED as is.
+  
+  2. **Otherwise, pull a hard-coded ED from a `match` statement**.
+  
+  3. If no matches are found, return `Balance::MAX` which will prune unknown
+     assets.
+    
+It should be updated as follows:
+
+  1. If an ED is defined for an asset in Asset Registry, return the ED as is.
+  
+  2. **If a asset ratio is defined for an asset in Asset Registry, calculate the
```

Comment:
how values would be initialized and released to devnet, testnet, kusama, and later to polkadot?

Reply:
I think it's good to have the list of asset exchange rates defined initially here for sure.

Reply:
Ideally, we could provide Asset Registry with a genesis config that would contain our initial assets and values. Otherwise, we'll need an approved origin to do this.

Reply:
Also, we may need another extrinsic in Asset Registry for creating assets with a defined asset ID rather than just getting the next value from Currency Factory.